src/main/java/org/mitre/synthea/world/agents/Provider.java

```diff
@@ -176,72 +194,121 @@ private static Provider findService(Person person,
    * 
    * @param state Name of the current state, ex "Massachusetts"
    * @param abbreviation State abbreviation, ex "MA"
    * @param filename Location of the file, relative to src/main/resources
    * @param servicesProvided Set of services provided by these facilities
    * @throws IOException if the file cannot be read
    */
-  public static void loadProviders(String state, String abbreviation, String filename,
+  public static void loadProviders(String state, String abbreviation, String filename, 
       Set<String> servicesProvided)
       throws IOException {
     String resource = Utilities.readResource(filename);
-    List<? extends Map<String,String>> csv = SimpleCSV.parse(resource);
-
-    for (Map<String,String> row : csv) {
+    Iterator<? extends Map<String,String>> csv = SimpleCSV.parseLineByLine(resource);
+    
+    while (csv.hasNext()) {
+      Map<String,String> row = csv.next();
       String currState = row.get("state");
-
+      
       // for now, only allow one state at a time
       if ((state == null)
           || (state != null && state.equalsIgnoreCase(currState))
           || (abbreviation != null && abbreviation.equalsIgnoreCase(currState))) {
+    
         Provider parsed = csvLineToProvider(row);
-
         parsed.servicesProvided.addAll(servicesProvided);
+        
         if ("Yes".equals(row.remove("emergency"))) {
           parsed.servicesProvided.add(Provider.EMERGENCY);
         }
-
+        
         // add any remaining columns we didn't explicitly map to first-class fields
         // into the attributes table
         for (Map.Entry<String, String> e : row.entrySet()) {
           parsed.attributes.put(e.getKey(), e.getValue());
         }
         
+        String city = parsed.city;
+        String address = parsed.address;
+
+        if (row.get("hasSpecialties") == null || row.get("hasSpecialties").equals("FALSE")) {
+          parsed.clinicianMap.put(ClinicianSpecialty.GENERAL_PRACTICE, 
+              parsed.generateClinicianList(1, ClinicianSpecialty.GENERAL_PRACTICE));
+        } else {
+          for (String specialty : ClinicianSpecialty.getSpecialties()) { 
+            if (row.get(specialty) != "") {
+              if (!row.get(specialty).equals("0")) {
+                parsed.clinicianMap.put(specialty, 
+                    parsed.generateClinicianList(Integer.parseInt(row.get(specialty)), specialty));
+              }
+            }
+          }
+          if (row.get(ClinicianSpecialty.GENERAL_PRACTICE).equals("0")) {
+            parsed.clinicianMap.put(ClinicianSpecialty.GENERAL_PRACTICE, 
+                parsed.generateClinicianList(1, ClinicianSpecialty.GENERAL_PRACTICE));
+          }
+        }
+        
+        
+        //TODO - determine how many clinicians based off the population
+
+        parsed.clinicianMap.put("GENERAL PRACTICE", 
+            parsed.generateClinicianList(1, "GENERAL PRACTICE"));
+
```

Comment:
Not sure this is necessary here - looks like both the `if` and the `else` blocks above set the General Practice value to at least 1.